product_uom_updatable/models/product_template.py

```diff
@@ -25,24 +24,28 @@ def write(self, vals):
 
     def _update_uom(self, uom_id, field_name):
         uom_obj = self.env["uom.uom"]
+        new_uom = uom_obj.browse(uom_id)
         sorted_items = sorted(self, key=lambda r: r[field_name])
+
         for key, products_group in groupby(sorted_items, key=lambda r: r[field_name]):
             product_ids = [p.id for p in products_group]
-            new_uom = uom_obj.browse(uom_id)
+
             if (
                 key.category_id == new_uom.category_id
                 and key.factor_inv == new_uom.factor_inv
             ):
                 # pylint: disable=sql-injection
-                query = sql.SQL(
-                    "UPDATE product_template SET {field} = %s WHERE id in %s"
-                ).format(field=sql.Identifier(field_name))
+                query = f"""
+                UPDATE product_template
+                SET {make_identifier(field_name)} = %s
```

Comment:
why not use SQL.identifier and according to the query in the source code, it is already wrapped by SQL object: https://github.com/odoo/odoo/blob/d122acdf7bb8c3d72090b2c5b116b8474945191b/odoo/tools/sql.py#L183